.github/workflows/tests.yml

```diff
@@ -21,8 +21,8 @@ jobs:
         run: git fetch origin ${GITHUB_REF}:ci-branch
       - name: "Check out branch"
         run: git checkout --force ci-branch
-      - name: "Run CMake to regenerate compile commands"
-        run: inv -r faasmcli/faasmcli dev.cmake --build Release
+      - name: "Build the targets to update compile commands and dependencies"
```

Comment:
`clang-tidy` is requiring an increasing amount of compilation targets. In particular, in this PR I encountered:
* `clang-tidy` requiring protobuf's `*.pb.*` compiled files.
* `clang-tidy` requiring `pistache` headers installed with `faabric`.

As one could expect this to be a recurring issue, and given that we are already building the targets in a, concurrent, build, we decided offline to build the `dev.tools` target in the `formatting` workflow as well.